Feature showing messages to the users, fixes #3799 (#4885)
ddev/ddev: docs/content/developers/remote-config.md

# Remote Config

DDEV supports downloading a remote config from the [`ddev/remote-config`](https://github.com/ddev/remote-config)
GitHub repository with messages that will be shown to the user. This feature
could be enhanced later with more information and configuration.

> Here is an example of a `remote-config.jsonc` file:

Reviewer: This might be better just as a link to the remote-config repo.

Author: Agree, on place less to maintain.

> # Remote config

Reviewer 2: Capitalize heading:

```suggestion
# Remote Config
```

> DDEV supports to download a remote config from the `ddev/remote-config` Github
> repository. At the moment the remote config contains messages which will be
> shown to the user see next chapter but can also be easily enhanced in the
> future with more information and configuration.

Reviewer 2: Shorten and link to repository:

```suggestion
DDEV supports downloading a remote config from the [`ddev/remote-config`](https://github.com/ddev/remote-config) GitHub repository with messages that will be shown to the user. This feature could be enhanced later with more information and configuration.
```

            ],
            "warnings": [
                {
                    "message": "Please update your installation as soon as possible, there is a big security risk by using this version.",
                    "conditions": [],
                    "versions": "<1.20"
                },
            ],
        },

        // One ticker messages is shown once in an interval and rotated to the
        // next afterwards
        "ticker": {
            "disabled": false,
            "interval": 5,
            "messages": [
                {
                    "message": "Did you know? You can restart your project at any time using `ddev restart`.",
                    "conditions": [],
                    "versions": ""
                },
                {
                    "message": "Did you know? You can open a browser heading to your project with `ddev launch`."
                },
            ]
        }
    }
}
```

## Messages

### Notifications

The defined messages are shown to the user every `interval` as long as not
`disabled`. Supported message types are `infos` and `warnings` where `infos`
are printed in a yellow box and `warnings` in a red box.

> The user can not influence notifications, they will be shown like defined in
> the `remote-config` repository.

Reviewer 2: ```suggestion
Messages will be shown as configured in the `remote-config` repository and the user cannot influence them.
```

### Ticker

> One of the defined messages is shown to the user every `interval` as long as
> not `disabled`, the messages are rotating.

Reviewer 2: Shorten, improve flow:

```suggestion
Messages rotate, with one shown to the user every `interval` as long as it’s not `disabled`.
```
